In [14]:
import pandas as pd
import numpy as np
from google.colab import files
files.upload()

{}

In [51]:
data = pd.read_csv('problem_set_1_data.csv')

In [53]:
data= data.rename(columns={"quizzes": "quiz_total", "attendedClass": "attended", "finalgrade": "final_grade","final_EXAM": "final_exam","student": "name"})
print(data.head())
print(data.tail())
print(data.describe())

   quiz_total  attended  final_grade  final_exam               name
0       78.50         1        99.38          46       Linnie Lietz
1       64.25         0        95.78          45      Mitch Mustain
2       76.75         0        97.42          41     Salina Chavera
3       69.00         1        92.50          38  Kimberely Conwell
4       64.00         0        92.50          39          Zack Burk
     quiz_total  attended  final_grade  final_exam                 name
166        73.5         0        96.25          45           Dana Malta
167        75.5         0        98.13          44         Sixta Heyden
168        41.0         0        69.38          32       Brianne Broome
169        77.0         0       101.25          47  Elouise Weatherholt
170        72.0         0        92.81          38        Addie Maharaj
       quiz_total    attended  final_grade  final_exam
count  171.000000  171.000000   171.000000  171.000000
mean    69.131579    0.192982    91.855380   39.67

The mean of quize:total is 69.13

The mean of attended is 0.19(which is meaningless)

The mean of final grade is 91.86

The mean of final exame is 39.67

Student is categorical variable thus does not have a mean.

In [63]:
boolean_index = data['attended'] == 1
print(boolean_index)

data_attended = data[boolean_index]
data_notattended = data[boolean_index== False]
print(data_attended)
print(data_notattended)
print("")

print(data_attended['quiz_total'].mean(),
data_attended['final_grade'].mean(),
data_attended['final_exam'].mean())

print(data_notattended['quiz_total'].mean(),
data_notattended['final_grade'].mean(),
data_notattended['final_exam'].mean())

0       True
1      False
2      False
3       True
4      False
       ...  
166    False
167    False
168    False
169    False
170    False
Name: attended, Length: 171, dtype: bool
     quiz_total  attended  final_grade  final_exam                 name
0         78.50         1        99.38          46         Linnie Lietz
3         69.00         1        92.50          38    Kimberely Conwell
5         63.00         1        89.38          36         Almeda Rizer
15        71.00         1        90.63          40          Mercy Dugal
16        77.00         1       101.25          46      Augustine Clair
27        75.00         1       101.25          45        Codi Whitcher
32        72.00         1        94.69          38           Nan Castro
36        73.00         1        87.81          33        Fern Iadarola
38        74.50         1        98.13          43        Delilah Starr
41        77.00         1       100.00          43         Maisie Monti
43        74.00         

THe average quiz total for those who attended class is higher than those who did not attended class(72.5> 68.33)

The average final grade for those who attended class is higher than those who did not attended class(95.07> 91.09)

THe average final_exame for those who attended class is slight higher than those who did not attended class(40.63> 39.44)

In [78]:
print(data[data.duplicated(['final_grade', "final_exam"])])
print(data.iloc[130,4])
print(data.iloc[139,4])


     quiz_total  attended  final_grade  final_exam                 name
52         70.0         0        90.00          40       Jana Satterlee
57         65.0         0        89.38          35         Winfred Haug
61         71.0         0        96.25          43        Laine Dowling
83         75.5         1       101.25          46          Tamica Wann
89         77.0         0        98.75          44     Kimbery Almaguer
104        75.0         0        97.50          46        Clark Gribble
110        72.5         0        97.81          44     Elfriede Gothard
130        67.5         0        95.94          43           Sade Byham
139        75.5         0        95.94          43    Arlette Hornbaker
140        72.5         0        90.00          35        Vincent Staub
151        77.0         1        93.75          40          Cristi Chae
152        69.0         0        95.00          42        Brock Crocket
156        71.0         0        95.31          41  Margarite Ka